projects/igniteui-angular/src/lib/data-operations/data-clone-strategy.ts

```diff
@@ -0,0 +1,13 @@
+import { cloneValue } from "../core/utils";
+
+export interface IDataCloneStrategy {
+    clone(data: any[]): any[];
+}
+
+export class DefaultDataCloneStrategy implements IDataCloneStrategy {
+    constructor() {  }
+
+    public clone(data: any[]): any[] {
```

Comment:
Does it always take `any[]`? Maybe make it generic so the end user can constraint types in their implementation.
Or `unknown` if this can't work